### PR TITLE
244 create webhook endpoint and server function for background checks

### PR DESCRIPTION
**Pull request by djwoun**
# Description
To test use Postman:
POST http://localhost:3000/api/bc-webhook/
Body: raw 
{
  "data": {
    "employee_email": "sophie.nguyen@example.com",
    "overall_status": "Passed"
  },
  "event_type": "overall_status_update"
}

## Relevant issue(s)
- #244 

## Questions
Is there a way to test with SpringVerify?

## Type of change
- [X] New feature (non-breaking change which adds functionality)

# Checklist:
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have assigned reviewers to this PR

### CODE REVIEW

File: src/types/dataModel/BackgroundCheckWebhookPayload.ts
Comment (zaviermiller) [inline src/types/dataModel/BackgroundCheckWebhookPayload.ts:1]: very minor change, but make the filename camelCase

File: src/server/actions/handleBackgroundCheckWebhook .ts
Comment (zaviermiller) [inline src/server/actions/handleBackgroundCheckWebhook .ts:1]: make this file just called `webhooks.ts`

File: src/server/actions/handleBackgroundCheckWebhook .ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    await dbConnect();
+    res = await VolunteerSchema.findOneAndUpdate(
+      { email: payload?.data.employee_email },
+      {
+        $set: {
+          'backgroundCheck.status': payload.data.overall_status,
+          'backgroundCheck.lastUpdated': new Date(),
+        },
+      },
+      { new: true }
```
Comment (zaviermiller) [inline src/server/actions/handleBackgroundCheckWebhook .ts:24]: dont create a new volunteer here

File: src/server/actions/handleBackgroundCheckWebhook .ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * @returns // volunteerschema or null
+ */
+export async function handleBackgroundCheckWebhook(
+  payload: BackgroundCheckWebhookPayload
+): Promise<void> {
+  let res;
+  try {
+    await dbConnect();
+    res = await VolunteerSchema.findOneAndUpdate(
+      { email: payload?.data.employee_email },
```
Comment (zaviermiller) [inline src/server/actions/handleBackgroundCheckWebhook .ts:17]: also find a volunteer that has a defined `backgroundCheck` field

File: src/server/actions/handleBackgroundCheckWebhook .ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      {
+        $set: {
+          'backgroundCheck.status': payload.data.overall_status,
+          'backgroundCheck.lastUpdated': new Date(),
```
Comment (zaviermiller) [inline src/server/actions/handleBackgroundCheckWebhook .ts:21]: dont update lastUpdate here, that is going to be used for CM staff to know the last time they performed a background check

**Review by zaviermiller — CHANGES_REQUESTED** [review]
minor changes

File: src/app/page.tsx [hunk 4]
```diff
@@ -1,5 +1,4 @@
 import { Typography } from '@mui/material';
-
```
Comment (zaviermiller) [inline src/app/page.tsx:2]: revert this change, just adds noise to pr

File: src/server/actions/webhooks.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+ * @param payload // BackgroundCheckWebhookPayload
+ * @returns // volunteerschema or null
+ */
+export async function handleBackgroundCheckWebhook(
+  payload: BackgroundCheckWebhookPayload
+): Promise<void> {
+  let res;
+  try {
+    await dbConnect();
+    res = await VolunteerSchema.findOneAndUpdate(
```
Comment (zaviermiller) [inline src/server/actions/webhooks.ts:16]: if res == null return 404

File: src/types/dataModel/backgroundCheckWebhookPayload.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { z } from 'zod';
+
+const zBackgroundCheckWebhookPayload = z.object({
+  data: z.object({
+    employee_email: z.string(),
+    overall_status: z.string(),
+  }),
+  event_type: z.string(),
```
Comment (zaviermiller) [inline src/types/dataModel/backgroundCheckWebhookPayload.ts:8]: also can you change this to be `z.literal("overall_status_update")`. I realized that springverify may add more webhooks in the future so we want to make sure we are only handling the overall status update ones.

File: src/types/dataModel/backgroundCheckWebhookPayload.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { z } from 'zod';
+
+const zBackgroundCheckWebhookPayload = z.object({
+  data: z.object({
+    employee_email: z.string(),
+    overall_status: z.string(),
```
Comment (zaviermiller) [inline src/types/dataModel/backgroundCheckWebhookPayload.ts:6]: also, can you make this field `z.union([z.literal("PASSED"), z.literal("FAILED")])` I am guessing the request from the webhook will have them as all uppercase since that was the example.

File: src/server/actions/webhooks.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+      {
+        email: payload?.data.employee_email,
+        backgroundCheck: { $exists: true },
+      },
+      {
+        $set: {
+          'backgroundCheck.status': payload.data.overall_status,
```
Comment (zaviermiller) [inline src/server/actions/webhooks.ts:23]: make sure you don't just set the value here, it may come back as "PASSED" or "FAILED". instead, first make it so that only the first letter is capitalized like how we are expecting it. i.e. "Passed" "Failed"

**Review by zaviermiller — CHANGES_REQUESTED** [review]

File: src/server/actions/webhooks.ts (hunk 5 above)
Comment (djwoun) [reply src/server/actions/webhooks.ts:16]: why?

Comment (djwoun) [reply src/server/actions/webhooks.ts:16]: oh I guess you meant to say return a 404 in the API if res == null

**Review by zaviermiller — APPROVED** [review]
LGTM
